### PR TITLE
try latest pycurl release

### PR DESCRIPTION
**Pull request by auvipy**
related https://github.com/celery/kombu/commit/f3b04558fa0df4ecc383c93e0b3b300d95e17c47

### CODE REVIEW

**Review by auvipy — COMMENTED** [review]
i think windows failures are expected. we need to use https://docs.pytest.org/en/6.2.x/skipping.html in the test for windows

**Comment by auvipy** [comment]
@hannseman what do think about my thoughts on windows build

**Comment by hannseman** [comment]
@auvipy I'm not that familiar with working on Python stuff in Windows but it looks like pycurl should have wheels for Windows so it should be installable. The error I'm looking at is this one:
```
  Collecting pycurl~=7.44.1
131
    Downloading pycurl-7.44.1.tar.gz (227 kB)
132
    Preparing metadata (setup.py): started
133
    Preparing metadata (setup.py): finished with status 'error'
134
    ERROR: Command errored out with exit status 10:
135
     command: 'D:\a\celery\celery\.tox\3.8-unit\Scripts\python.EXE' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pip-install-innclb3y\\pycurl_875358ca3df246c5b05c8fef321401cb\\setup.py'"'"'; __file__='"'"'C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pip-install-innclb3y\\pycurl_875358ca3df246c5b05c8fef321401cb\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base 'C:\Users\runneradmin\AppData\Local\Temp\pip-pip-egg-info-cy3b6h_w'
136
         cwd: C:\Users\runneradmin\AppData\Local\Temp\pip-install-innclb3y\pycurl_875358ca3df246c5b05c8fef321401cb\
137
    Complete output (1 lines):
138
    Please specify --curl-dir=/path/to/built/libcurl
139
```

From https://github.com/celery/celery/runs/4165002584?check_suite_focus=true

It looks like the  7.44.1 does not have any wheels, the latest one I can find which do is 7.43.05 (https://pypi.org/project/pycurl/7.43.0.5/#files). Also see this issue https://github.com/pycurl/pycurl/issues/710.

**Comment by auvipy** [comment]
we officially don't support windows so we are safe on that. I have also added a comment there to release a wheel for latest release

**Comment by hannseman** [comment]
> we officially don't support windows so we are safe on that. I have also added a comment there to release a wheel for latest release

Looks like there's no immediate plan to bring back wheels to pycurl, see https://github.com/pycurl/pycurl/issues/657

If the intention is to drop the pin of pycurl to the wheeled version (as already done in kombu) one could look into specifying `--curl-dir` on windows runs in CI. See https://github.com/pycurl/pycurl/blob/master/INSTALL.rst#windows

**Comment by auvipy** [comment]
@graingert we don't support windows 32 bit should we follow the --curl-dir path for windows build for latest pycurl release?

File: requirements/test-ci-default.txt [hunk 1]
```diff
@@ -20,4 +20,4 @@
 -r extras/azureblockblob.txt
 
 # SQS dependencies other than boto
-pycurl==7.43.0.5  # Latest version with wheel built (for appveyor)
+pycurl~=7.44.1  # Latest version with wheel build
```
Comment (auvipy) [inline requirements/test-ci-default.txt:23]:
```suggestion
pycurl>=7.43.0.5; sys_platform != 'win32'
```

File: requirements/extras/sqs.txt [hunk 2]
```diff
@@ -1 +1,3 @@
-kombu[sqs]
+boto3>=1.9.12
+pycurl~=7.44.1  # Latest pypi release without wheels provided
```
Comment (auvipy) [inline requirements/extras/sqs.txt:2]:
```suggestion
pycurl>=7.43.0.5; sys_platform != 'win32'
```

File: requirements/extras/sqs.txt [hunk 3]
```diff
@@ -1 +1,5 @@
-kombu[sqs]~=5.3.0b3
+boto3>=1.9.12
+pycurl>=7.43.0.5; sys_platform != 'win32'
```
Comment (auvipy) [inline requirements/extras/sqs.txt:2]: this seems to working by not running on windows

File: requirements/test-ci-default.txt [hunk 4]
```diff
@@ -21,4 +21,5 @@
 git+https://github.com/celery/kombu.git
 
 # SQS dependencies other than boto
-pycurl==7.43.0.5
+pycurl>=7.43.0.5; sys_platform != 'win32'
```
Comment (auvipy) [inline requirements/test-ci-default.txt:24]:
```suggestion
pycurl>=7.43.0.5; sys_platform != 'win32'; platform_python_implementation=="CPython"
```

File: requirements/extras/sqs.txt (hunk 3 above)
Comment (auvipy) [inline requirements/extras/sqs.txt:2]:
```suggestion
pycurl>=7.43.0.5; sys_platform != 'win32'; platform_python_implementation=="CPython"
```

**Comment by auvipy** [comment]
couchbase>=3.0.0; platform_python_implementation!='PyPy' and (platform_system != 'Windows' or python_version < '3.10') --- got this and will adjust accordingly

File: requirements/extras/sqs.txt [hunk 5]
```diff
@@ -1 +1,5 @@
-kombu[sqs]~=5.3.0b3
+boto3>=1.9.12
+pycurl>=7.43.0.5; sys_platform != 'win32'; platform_python_implementation=="CPython"
```
Comment (auvipy) [inline requirements/extras/sqs.txt:2]:
```suggestion
pycurl>=7.43.0.5; sys_platform != 'win32' and platform_python_implementation=="CPython"
```

File: requirements/test-ci-default.txt [hunk 6]
```diff
@@ -21,4 +21,5 @@
 git+https://github.com/celery/kombu.git
 
 # SQS dependencies other than boto
-pycurl==7.43.0.5
+pycurl>=7.43.0.5; sys_platform != 'win32'; platform_python_implementation=="CPython"
```
Comment (auvipy) [inline requirements/test-ci-default.txt:24]:
```suggestion
pycurl>=7.43.0.5; sys_platform != 'win32' and platform_python_implementation=="CPython"
```

**Review by auvipy — COMMENTED** [review]
merging as is

File: requirements/extras/sqs.txt [hunk 7]
```diff
@@ -1 +1,4 @@
+boto3>=1.26.143
+pycurl>=7.43.0.5; sys_platform != 'win32' and platform_python_implementation=="CPython"
```
Comment (sparrowt) [inline requirements/extras/sqs.txt:2]: is pycurl no longer required for sqs on windows?

Comment (sparrowt) [reply requirements/extras/sqs.txt:2]: also why do these 3 need duplicating in here when they're already in https://github.com/celery/kombu/blob/main/requirements/extras/sqs.txt

Comment (auvipy) [reply requirements/extras/sqs.txt:2]: we just don't test it on windows due to lack of wheels

**Pull request merged**